src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> node;
+    private int size;
+    private Node<T> head = null;
+    private Node<T> tail = null;
+
     @Override
     public void add(T value) {
+        node = new Node<>(null, value,null);
+        if (head == null) {
+            head = tail = node;
+            size++;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            node.next = null;
+            tail = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        throwIndexOutOfBoundsAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> current = findNodeByIndex(index);
+        Node<T> newNode = new Node<>(current.prev, value, current);
+        current.prev = (current.prev != null) ? current.prev.next = newNode : (head = newNode);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            node = new Node<>(null, list.get(i), null);
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+            size++;
+        }
```

Comment:
You can simplify the `addAll(List<T> list)` method by calling the `add(T value)` method inside the for loop.